drivers/interrupt_controller/intc_miwu.c

```diff
@@ -215,6 +290,9 @@ int npcx_miwu_interrupt_configure(const struct npcx_wui *wui,
 			NPCX_WKEDG(base, wui->group) |= pmask;
 			break;
 		default:
+#ifdef CONFIG_NPCX_MIWU_BOTH_EDGE_TRIG_WORKAROUND
+			k_spin_unlock(&data->lock, key);
```

Comment:
Non-blocking comment - I prefer a goto to handle common cleanup actions.  This would also eliminate some ifdefs.

```
    ret = -EINVAL;
    goto early_exit;
...
early_exit:
#ifdef WORKAROUND
    k_spin_unlock();
#endif
return ret;

Reply:
Done.